plugins/module_utils/vm_snapshot.py

```diff
@@ -232,3 +307,83 @@ def send_delete_request(
         return rest_client.delete_record(
             f"/rest/v1/VirDomainSnapshot/{snapshot_uuid}", False
         )
+
+    @classmethod
+    # Used to rename dict keys of a hypercore object that doesn't have an implemented class
+    def hypercore_block_device_to_ansible(
+        cls, _dict: Optional[Dict[Any, Any]]
+    ) -> Optional[Dict[Any, Any]]:
+        if _dict is None:
+            return None
+
+        new_dict = dict()
+        re_pattern_first = re.compile("(.)([A-Z][a-z]+)")
+        re_pattern_second = re.compile("([a-z0-9])([A-Z])")
+
+        for key in _dict.keys():
+            new_key = re.sub(re_pattern_first, r"\1_\2", key)
+            new_key = re.sub(re_pattern_second, r"\1_\2", new_key).lower()
+
+            if new_key == "uuid":
+                new_key = "block_device_uuid"
+            if new_key == "vir_domain_uuid":
+                new_key = "vm_uuid"
+
+            new_dict[new_key] = _dict[key]
+        return new_dict
+
+    @classmethod
+    def get_vm_disk_info_by_uuid(
```

Comment:
Hmmm... to get block device info only, I think it is easier to use ``rest/v1/virDomainBlockDevice/uuid`` otherwise another filter function would be needed to filter results returned by ``rest/v1/VirDomain/uuid``. For the last function in this file (``get_external_vm_uuid``) which does actually use the endpoint ``rest/v1/VirDomain/uuid`` I guess I could've just used the VM class instead... Well, as you've said, for now I will leave this code as is, and then later if needed, I will change it for optimization.